client/securedrop_client/config.py

```diff
@@ -1,31 +1,59 @@
-import json
 import logging
 import os
+from collections.abc import Generator
+from contextlib import contextmanager
 
 logger = logging.getLogger(__name__)
 
 
+@contextmanager
+def try_qubesdb() -> Generator:
+    """Minimal context manager around QubesDB() → QubesDB.close() when
+    available."""
+    db: bool | "QubesDB" = False
+
+    try:
+        from qubesdb import QubesDB
+
+        db = QubesDB()
+        yield db
+
+    except ImportError:
+        yield db
+
+    finally:
+        if db:
+            db.close()  # type: ignore[union-attr]
+
+
 class Config:
-    CONFIG_NAME = "config.json"
+    """Configuration loaded at runtime from QubesDB (if available) or
+    environment variables."""
 
-    def __init__(self, journalist_key_fingerprint: str) -> None:
-        self.journalist_key_fingerprint = journalist_key_fingerprint
+    # Mapping of `Config` attributes (keys) to how to look them up (values)
+    # from either QubesDB or the environment.
+    mapping = {
+        "gpg_domain": "QUBES_GPG_DOMAIN",
+        "journalist_key_fingerprint": "SD_SUBMISSION_KEY_FPR",
+    }
 
     @classmethod
-    def from_home_dir(cls, sdc_home: str) -> "Config":
-        full_path = os.path.join(sdc_home, Config.CONFIG_NAME)
-
-        try:
-            with open(full_path) as f:
-                json_config = json.loads(f.read())
-        except Exception as e:
-            logger.error(f"Error opening config file at {full_path}: {e}")
-            json_config = {}
-
-        return Config(
-            journalist_key_fingerprint=json_config.get("journalist_key_fingerprint", None)
-        )
-
-    @property
-    def is_valid(self) -> bool:
-        return self.journalist_key_fingerprint is not None
+    def load(self) -> "Config":
+        """For each attribute, look it up from either QubesDB or the environment."""
+        config = Config()
+
+        with try_qubesdb() as db:
+            for store, lookup in self.mapping.items():
+                if db:
+                    logger.debug(f"Reading {lookup} from QubesDB")
+                    value = db.read(f"/vm-config/{lookup}")
+                    if len(value) > 0:
+                        value = None
```

Comment:
Note to self: We should handle this case the same way here as in:

https://github.com/freedomofpress/securedrop-client/blob/f6ad518261db5ddd4c1fe7cda9ee1cfb090c49dc/proxy/src/config_qubesdb.rs#L24-L26

Reply:
Done in d8c02e3d5a92cd098f732bfb1abead8595c9a935.